File: core/src/org/tetawex/ecf/actor/TimerActor.java
package org.tetawex.ecf.actor;

import com.badlogic.gdx.scenes.scene2d.Actor;

/**
 * Created by Tetawex on 06.06.2017.
 */
public class TimerActor  extends Actor{
    public interface TimerActorListener{
        void onTimePassed();
    }
    private float timePassed=0f;
    private float timeSpan=0f;

    private TimerActorListener listener;
    public TimerActor(float timeSpan, TimerActorListener listener){
        this.listener=listener;
        this.timeSpan=timeSpan;
    }
    @Override
    public void act(float deltaTime) {
        timePassed+=deltaTime;
        if(timePassed>=timeSpan)
            listener.onTimePassed();
    }
}

File: core/src/org/tetawex/ecf/presentation/widget/TimerActor.kt
package org.tetawex.ecf.presentation.widget

import com.badlogic.gdx.scenes.scene2d.Actor

/**
 * Created by Tetawex on 06.06.2017.
 */
class TimerActor(val timeSpan: Float, private val listener: TimerActorListener) : Actor() {

    private var timePassed = 0f

    interface TimerActorListener {
        fun onTimePassed()
    }

    override fun act(deltaTime: Float) {
        timePassed += deltaTime
        if (timePassed >= timeSpan)
            listener.onTimePassed()
    }
}
